flexget/components/sites/sites/filelist_api.py

```diff
@@ -0,0 +1,180 @@
+        403: 'Invalid passkey/username',
+        429: 'Rate limit reached',
+        503: 'Service unavailable',
+    }
+
+    valid_categories = {
+        'Anime': '24',
+        'Audio': '11',
+        'Desene': '15',
+        'Diverse': '18',
+        'Docs': '16',
+        'FLAC': '5',
+        'Filme 3D': '25',
+        'Filme 4K': '6',
+        'Filme 4K Blu-Ray': '26',
+        'Filme Blu-Ray': '20',
+        'Filme DVD': '2',
+        'Filme DVD-RO': '3',
+        'Filme HD': '4',
+        'Filme HD-RO': '19',
+        'Filme SD': '1',
+        'Jocuri Console': '10',
+        'Jocuri PC': '9',
+        'Linux': '17',
+        'Mobile': '22',
+        'Programe': '8',
+        'Seriale 4K': '27',
+        'Seriale HD': '21',
+        'Seriale SD': '23',
+        'Sport': '13',
+        'TV': '14',
+        'Videoclip': '12',
+        'XXX': '7',
+    }
+
+    valid_extras = {'internal': 1, 'moderated': 1, 'freeleech': 1, 'doubleup': 1}
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'passkey': {'type': 'string'},
+            'category': one_or_more({'type': 'string', 'enum': list(valid_categories.keys())}),
+            'internal': {'type': 'boolean', 'default': False},
+            'moderated': {'type': 'boolean', 'default': False},
+            'freeleech': {'type': 'boolean', 'default': False},
+            'doubleup': {'type': 'boolean', 'default': False},
+        },
+        'required': ['username', 'passkey'],
+        'additionalProperties': False,
+    }
+
+    def get(self, url, params):
+
+        try:
+            response = requests.get(url, params=params, raise_status=False)
+        except RequestException as e:
+            raise plugin.PluginError(f'FileList request failed badly! {e}')
+
+        if response.status_code in self.error_codes.keys():
+            # get error message if we have one, else use generic
+            error_message = response.json().get('error', self.error_codes[response.status_code])
+
+            raise plugin.PluginError(
+                f'FileList request failed; err code: {response.status_code}; err msg: `{error_message}`'
+            )
+
+        return response
+
+    def dump_entry(self, entry):
+        entry_items = {key: value for key, value in entry.items()}
+        logger.debug(f'entry_items={entry_items}')
+
+    @plugin.internet(logger)
+    def search(self, task, entry, config):
+        """
+            Search for entries on FileList
+        """
+        entries = list()
+
+        # TODO remove later
+        self.dump_entry(entry)
+
+        # mandatory params
+        params = {'username': config['username'], 'passkey': config['passkey'], 'action': 'search-torrents'}
+
+        # category
+        if config.get('category'):
+            params['category'] = (
+                ','.join([self.valid_categories[cat] for cat in config['category']])
```

Comment:
You can pass a generator to join, no need for the list comprehension here

Reply:
`','.join(self.valid_categories[cat] for cat in config['category'])` you mean like this ? TIL